### ISSUE TITLE
toot gallery doesn't work on static pages

### ISSUE THREAD

**Issue by MightyPork** [glitch-soc/mastodon#306]
test: https://dev.glitch.social/@MightyPork/99313184729397777

probably bc it isn't static anymore :roll_eyes: 

```
public-82ba25d5379b30bf6ec3.js:1 TypeError: Cannot read property 'max_toot_chars' of null
    at Object.<anonymous> (public-82ba25d5379b30bf6ec3.js:1)
    at e (locales-ce34382215c03a9d5751.js:1)
    at Object.<anonymous> (public-82ba25d5379b30bf6ec3.js:1)
    at e (locales-ce34382215c03a9d5751.js:1)
    at r (public-82ba25d5379b30bf6ec3.js:1)
    at <anonymous>
```

**Comment by beatrix-bitrot** [comment]
hnnnnnghhhh static pages have no reason to contain such compose related stuff...

**Comment by yipdw** [comment]
The error occurs because:

1. the `public` pack contains
```js
const emojify = require('flavours/glitch/util/emoji').default;
```
2. `flavours/glitch/util/emoji/index.js` contains
```js
import { autoPlayGif } from 'flavours/glitch/util/initial_state';
```

We could put the initial state JSON in `accounts/show`; it's already done on several other pages like it, e.g.
```
about/show.html.haml:  %script#initial-state{ type: 'application/json' }!= json_escape(@initial_state_json)
```

Regardless of that change, `flavours/glitch/util/initial_state` needs to be modified to handle the case where the JSON doesn't exist (because it won't on unauthenticated accesses).  The `getMeta` function does this; `max_toot_chars` doesn't.  We can fix that by checking for `initialState` presence there also.

There is a point that `max_toot_chars` and `autoPlayGif` probably don't belong in the same object, but it's not clear to me how we would separate those.  In some way they _are_ both part of the app state.

**Comment by yipdw** [comment]
2dc4fbbd1a13adbdca26a0b42ee6e289e736bba1 gets the gallery working again, but I am not sure if it is the ultimate fix we want.  We could, for example, do any of the following:

- put the initial state in `accounts/show` also
- separate `max_chars` from `initial_state`

I guess that's more of a @marrus-sh question?

**Comment by yipdw** [comment]
The image on https://dev.glitch.social/@MightyPork/99313184729397777 now shows up, both with and without authentication, so I'm going to mark this issue as closed.  However, I'd still like to hear about future directions as indicated in my previous comment; please feel free to respond.

**Comment by marrus-sh** [comment]
I kind of like this approach actually :P If the initial state isn't provided, I don't have a problem with `initial_state.js` providing defaults (it's better than throwing an error anyway) and i don't know where else you would put `max_chars` if not in there. The fact that we export all the properties individually makes it a bit wordy but I think we'll just have to live with it (one alternate solution would be to move the `const maxChars = initialState && initialState.max_chars || 500` into the `<Composer>` component or w/e but i'm not sure that's any better)
